synce_dev.c

```diff
@@ -208,47 +231,63 @@ static int port_set_ql_line_input(struct synce_dev *dev,
 
 static int update_ql_line_input(struct synce_dev *dev)
 {
-	struct synce_port *p, *best_p = dev->best_source;
+	struct synce_port *p, *best_p;
+	struct synce_clock_source *c;
 	int ret = 0;
 
-	LIST_FOREACH(p, &dev->ports, list) {
-		if (dev->d_state == EEC_HOLDOVER) {
-			pr_debug("act on EEC_HOLDOVER for %s",
-				 synce_port_get_name(p));
-			ret = port_set_dnu(p, dev->extended_tlv);
-			if (ret) {
-				pr_err("%s set DNU failed on %s",
-				       __func__, dev->name);
-				return ret;
-			}
-		} else if ((dev->d_state == EEC_LOCKED ||
-			   dev->d_state == EEC_LOCKED_HO_ACQ) && best_p) {
-			pr_debug("act on EEC_LOCKED/EEC_LOCKED_HO_ACQ for %s",
-				 synce_port_get_name(p));
-			/* on best port send DNU, all the others
-			 * propagate what came from best source
-			 */
-			if (p != best_p) {
-				ret = port_set_ql_line_input(dev, p,
-								 best_p);
-			} else {
-				ret = port_set_dnu(p, dev->extended_tlv);
-			}
+	best_p = dev->best_source ? dev->best_source->port : NULL;
 
-			if (ret) {
-				pr_err("%s set failed on %s",
-				       __func__, dev->name);
-				return ret;
+	LIST_FOREACH(c, &dev->clock_sources, list) {
+		if (c->type == PORT ) {
+			p = c->port;
+			if (dev->d_state == EEC_HOLDOVER) {
+				pr_debug("act on EEC_HOLDOVER for %s",
+					 synce_port_get_name(p));
+				ret = port_set_dnu(p, dev->extended_tlv);
+				if (ret) {
+					pr_err("%s set DNU failed on %s",
+					       __func__, dev->name);
+					return ret;
+				}
+			} else if ((dev->d_state == EEC_LOCKED ||
+				   dev->d_state == EEC_LOCKED_HO_ACQ) &&
```

Comment:
dev should start in the column as dev in previous line

Reply:
fixed